netbox_agent/server.py

```diff
@@ -79,7 +79,11 @@ def update_netbox_location(self, server):
             update = True
             server.site = nb_dc.id
 
-        if rack and server.rack and server.rack.id != nb_rack.id:
+        if (
+            rack
+            and all(map(lambda x: x is not None and "id" in x, (nb_rack, server.rack)))
```

Comment:
Seems overly complicated, why not just something like this?

```suggestion
            and server.rack
            and nb_rack
```